theme-utils.mjs

```diff
@@ -34,13 +34,166 @@ const coreThemes = ['twentyten', 'twentyeleven', 'twentytwelve', 'twentythirteen
+			{
+				helpText = 'Uses rsync to copy all modified files for any modified themes from the local machine to your sandbox.';
+				break;
+			}
+		case "push-theme-to-sandbox":
+			{
+				additionalArgs = '<theme-slug>';
+				helpText = 'Uses rsync to copy all modified files for the specified theme from the local machine to your sandbox.';
+				break;
+			}
+		case "push-premium-to-sandbox":
+			{
+				helpText = 'Uses rsync to copy all modified files for all premium themes from the local machine to your sandbox. For the blockbase theme, all instances of the string "blockbase" are replaced with "blockbase-premium".';
+				break;
+			}
+		case "version-bump-themes":
+			{
+				helpText = 'Bump the version of any theme that has had changes since the last deployment. This includes bumping the version of any parent themes and updating the changelog for the theme.';
+				break;
+			}
+		case "land-diff":
+			{
+				helpText = '';
+				break;
+			}
+		case "deploy-preview":
+			{
+				additionalArgs = '<arc diff id>';
+				helpText = 'Run arc land to merge in the specified diff id.';
+				break;
+			}
+		case "deploy-theme":
+			{
+				additionalArgs = '<array of theme slugs>';
+				helpText = 'This runs "deploy pub <theme>" on the provided list of themes.';
+				break;
+			}
+		case "build-com-zip":
+			{
+				additionalArgs = '<theme-slug>';
+				helpText = 'Build the production zip file for the specified theme.';
+				break;
+			}
+		case "pull-core-themes":
+			{
+				helpText = 'Use rsync to copy any changed public CORE theme files from your sandbox to your local machine. CORE themes are any of the Twenty<whatever> themes.';
+				break;
+			}
+		case "push-core-themes":
+			{
+				helpText = 'Use rsync to copy any changed public CORE theme files from your local machine to your sandbox. CORE themes are any of the Twenty<whatever> themes.';
+				break;
+			}
+		case "sync-core-theme":
+			{
+				additionalArgs = '<theme-slug> <since-revision>';
+				helpText = 'Given a theme slug and SVN revision, sync the theme from the specified revision to the latest. This command is generally run by deploy-sync-core-theme and not by itself.';
+				break;
+			}
+		case "deploy-sync-core-theme":
+			{
+				additionalArgs = '<theme-slug> <since-revision>';
+				helpText = 'Given a theme slug and SVN revision, sync the theme from the specified revision to the latest. This command contains additional prompts and error checking not provided by sync-core-theme.';
+				break;
+			}
+		case "update-theme-changelog":
+			{
+				additionalArgs = '<theme-slug> <add-changes, true/false>'
+				helpText = 'Use the commit log to build a list of recent changes and add them as a new changelog entry. If add-changes is true, the updated readme.txt will be staged.';
+				break;
+			}
+		case "rebuild-theme-changelog":
+			{
+				additionalArgs = '<theme-slug> <since>';
+				helpText = 'Rebuild the entire change long from the given starting hash.';
+				break;
+			}
```

Comment:
This is super helpful.

I have the following thoughts in the view of maintaining all commands from a single place.

Turn the information into an object. (also considering the switch case in line 17-38)
```js
const commands = {
  "push-button-deploy": { helpText: '',  additionalArgs: '', run: () =>  pushButtonDeploy() },
  "deploy-sync-core-theme": { helpText: '',  additionalArgs: '', run: () =>  deploySyncCoreTheme(args?.[1], args?.[2]) },
  ...
}
```

now the switch cases could be replaced by
```
// line 17-38
commands[command].run();

// line 75-190
const {helpText, additionalArgs} = commands[command];
```

Also `showHelp` method can be simplified to

```js
return `Available commands` + Object.keys(commands).join('\n')
```

Reply:
I like that! Much tidier looking!

Thanks for the suggestion, I should be able to get that taken care of today.